### PR TITLE
Allow distributed breakpoint to skip the first few calls

### PR DESCRIPTION
**Pull request by fegin**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #129511

Summary:
PDB allows to do conditional breakpoint but the ability won't work in the distributed environment. We can still do conditional breakpoint by doing the following:

```
counter = 0

global counter
count += 1
if counter > 100:
  dist.breakpoint()
```

This PR makes dist.breakpoint() support this feature as a syntax sugar.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @tianyu-l @yf225 @chauhang

### CODE REVIEW

File: torch/distributed/__init__.py [hunk 1]
```diff
@@ -73,14 +75,25 @@ def interaction(self, *args, **kwargs):
...
         Args:
             rank (int): Which rank to break on.  Default: ``0``
+            skip (int): Skip the first ``skip`` calls to this breakpoint. Default: ``0``.
         """
+        if skip > 0:
+            key = hash(str(traceback.format_exc()))
+            counter = _breakpoint_cache.get(key, 0) + 1
+            _breakpoint_cache[key] = counter
+            if counter <= skip:
+                print(f"Skip the breakpoint, {counter=}.")
```
Comment (wconstab) [inline torch/distributed/__init__.py:94]: logger.debug? or .info maybe? but probably not print?

**Review by wconstab — APPROVED** [review]
lgtm with one nit for printing. thanks for adding this!

**Review by c-p-i-o — APPROVED** [review]

**Pull request closed without merging**